Create a skeleton of a Django project

slsh1o opened the pull request:
Init a first Django app and fix all raised by linters issue.

Need this commits on master branch to proceed with tutorial.

Ask for review also: @BogdanShevchenkoo @ZVitaliy @dark-lord671

slsh1o commented:
@webknjaz ahh, yes. As you've said build failed.
Need help with it.
Received this error on travis-ci:
```console
pylint...................................................................Failed
- hook id: pylint
- exit code: 1
Executable `pylint` not found
```

webknjaz commented:
@slsh1o the current setup doesn't install pylint. `local` means that it hits the current env where `pre-commit` itself is installed so it should contain `pylint` too. I'd like to see that change in a separate PR.

webknjaz commented:
> Also some additional ignorecase and args was added to .gitignore and

This is not really related to *adding a skeleton*, is it? If you want it, put it in a separate PR. Although, I don't really like to customize `gitignore.io` template because if you want to update it from upstream you may lose extra changes. Are you sure you can't generate a template that contains it in that service?

> .pre-commit-config.yaml to fix C0415, C0103 and E501.

So here's the thing: before this commit, those errors don't exist, so applying this commit does not fix them in master as you claim it does. Which makes it something that doesn't add any value but adds confusion because you've created a commit that's already clean and there's no way to actually see those errors that you claim to exist.

webknjaz commented:
> some additional ignorecase and args was added

Also, this is totally non-concrete: how am I supposed to guess what you changed by reading this?

webknjaz commented on .pre-commit-config.yaml:
```diff
@@ -48,5 +48,7 @@ repos:
         name: pylint
         entry: pylint
+        args:
+        - --good-names=urlpatterns
```
This is a good case when to change pylint's rules: when there's something that the framework you use enforces, you should respect those principles. Well done!

Nowadays I lean towards creating a proper config file for pylint rather than using args but it's fine for now, it can be done separately.

Another thing to try is https://pypi.org/project/pylint-django/ — I believe that using it will silence a number of default pytest rules that conflict with the django way of doing things.

webknjaz commented on manage.py:
```diff
@@ -0,0 +1,23 @@
+    """Invoke main function."""
+    os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'mysite.settings')
+    try:
+        # pylint: disable=C0415
```
You can also use more descriptive rule identifiers rather than just numeric IDs. It's easier to understand what the rule implements w/o having to google it:
```suggestion
        # pylint: disable=import-outside-toplevel
```

webknjaz replied:
Still, I'm not sure why this message appears here. Maybe it could go away by just installing django rather than trying to ignore things.

webknjaz commented on mysite/__init__.py:
```diff
@@ -0,0 +1 @@
+"""Django website skeleton."""
```
Well, I think the skeleton is imprecise. While it is one, it's just a temporary state. Here it's just a package. So I'd drop that last word now.

webknjaz commented on mysite/asgi.py:
```diff
@@ -0,0 +1,16 @@
+"""
+ASGI config for mysite project.
+
+It exposes the ASGI callable as a module-level variable named ``application``.
```
I don't think that the framework documentation belongs in docstrings. The generator put it here for your convenience. It's useful for you as a getting started pointer but it's useless in the context of the concrete project as it's about implementing a website, not putting links to django docs everywhere.

webknjaz commented on mysite/settings.py:
```diff
@@ -0,0 +1,133 @@
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/3.0/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
```
Make sure you understand what this implies.

webknjaz commented on mysite/settings.py:
```diff
@@ -0,0 +1,133 @@
+    },
+    {
+        'NAME': (
+            'django.contrib.auth.password_validation'
```
I bet you're missing a dot here.

webknjaz reviewed (changes requested):
Plz apply the comments I wrote to all the similar places. I didn't copy each comment to every place where the same thing needs to be adjusted in the same manner.

Commit:
Add pylint args in pre-commit-config to fit Django

For further commits w/ Django pylint_django plugin will help pylint to
correctly works w/ Django variable names but urlpatterns var name
should be passed here explicitly w/ good-names arg because pylint_django
can't help with it.

Commit:
Init the skeleton of a Django project

Remove all redundant info from docsings generated by Django by default
and add absent one in __init__.py and manage.py to fit pep8.

Also set STATIC_ROOT and TIME_ZONE in settings.py to proceed w/
djangogirls tutorial.

slsh1o commented:
Ok. Another try from scratch but on the same PR :)

webknjaz commented on .pre-commit-config.yaml:
```diff
@@ -1,6 +1,6 @@
 repos:
 -   repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v2.5.0  # Use the ref you want to point at
```
This change is unrelated to the purpose of this PR and should be submitted separately.

webknjaz commented on .pre-commit-config.yaml:
```diff
@@ -36,7 +36,7 @@ repos:
     -   id: remove-crlf
     -   id: forbid-tabs
     -   id: remove-tabs
-        args: [ --whitespaces-count, '4' ]  # defaults to: 4
```
This change is unrelated to the purpose of this PR and should be submitted separately.

webknjaz commented on .pre-commit-config.yaml:
```diff
@@ -48,5 +48,8 @@ repos:
         name: pylint
         entry: pylint
+        args:
+        -   --load-plugins=pylint_django
```
The spacing is weird, it should be smth like
```suggestion
        - --load-plugins=pylint_django
        - --good-names=urlpatterns
```

webknjaz commented on mysite/asgi.py:
```diff
@@ -0,0 +1,9 @@
+"""ASGI config for mysite project."""
```
It's not a config but an application init.

webknjaz commented on mysite/settings.py:
```diff
@@ -0,0 +1,112 @@
+
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+SECRET_KEY = os.getenv('KPI_SECRET_KEY')
```
Good point, I'll post a task about improving the deployment config even more later.

webknjaz commented on mysite/urls.py:
```diff
@@ -0,0 +1,8 @@
+"""The mysite URL Configuration."""
```
I'd say it's rather a mapping. But config is also ok. Except that it's weird to see it capitalized.

webknjaz commented on mysite/wsgi.py:
```diff
@@ -0,0 +1,9 @@
+"""WSGI config for mysite project."""
```
```suggestion
"""WSGI application initialization for mysite project."""
```

webknjaz reviewed (changes requested):
Plz move things that aren't directly related to the PR purpose to a separate PR.
Also, the gating is set up to disallow merges that don't have green CI status, just like in real projects. So when I approve this PR you'll be unable to merge if it's red because merging it should turn the project from one working state into another working state. That said, fixing the lint setup in CI by installing an extra package may be considered as a thing deserving its own PR. But adding an extra plugin that improves linting of things added here (like Django-specific rules) probably belongs here.

P.S. You have a typo in one of the commit descriptions:
```diff
-becouse
+because
```

slsh1o commented:
Now should be ok but I still need #24 on master to pass Travis test here.

webknjaz commented:
Make sure to close&re-open this PR after merging #24. This will re-trigger CI builds.

slsh1o committed:
Merge branch 'master' into features/init-django-project

webknjaz commented:
Plz don't make foxtrot merges
